shell/platform/android/io/flutter/plugin/common/PluginRegistry.java

```diff
@@ -201,4 +211,12 @@
     interface ViewDestroyListener {
         boolean onViewDestroy(FlutterNativeView view);
     }
+
+    /**
+     * When implemented, this interface can be provided to a background service
```

Comment:
This javadoc is not clear to me. It sounds to me as if any implementation is supposed to look like this:
```java
class MyPluginRegistrant implements PluginRegistrant {
  @Override
  public void register(PluginRegistry registry) {
    GeneratedPluginRegistrant.registerWith(registry);
  }
}
```
I'm probably wrong, though. What stops a background service from calling `GeneratedPluginRegistrant.registerWith` directly?

1. I think the javadoc for interface `Xxx` should start by stating what an `Xxx` is. Then it can comment on specific use cases as applicable.
1. It seems the `PluginRegistrant` concept might be used also in contexts unrelated to background services, and we should avoid tying the two concepts together in the javadoc.
1. It's a bit odd that the `GeneratedPluginRegistrant` is not actually a `PluginRegistrant`. Maybe this interface should be renamed. Or its `register` method could be renamed `registerWith` to get at least some consistency.

Reply:
Tried to improve this. Renamed interface PluginRegistrant to interface PluginRegistrantCallback. A background service in a plugin doesn't have direct access to an application's GeneratedPluginRegistrant.